### ISSUE TITLE
server crashes when user tries to log into ejabberd version 16.12.11

### ISSUE THREAD

**Issue by sudoman** [processone/ejabberd#1487]
> What version of ejabberd are you using?

16.12.11

> What operating system (version) are you using?

Trisquel 8

> How did you install ejabberd (source, package, distribution)?

Source. I imported the data from an older server, approximately version 2.xx. Some data import methods would cause the import process to crash.

> What did not work as expected? Are there error messages in the log? What
> was the unexpected behavior? What was the expected result?

Using the non-crashing import method, the server would then crash when attempting to log in.

````
2017-01-03 00:37:45 =ERROR REPORT====
** State machine <0.754.0> terminating 
** Last message in was {route,{jid,<<"user_x">>,<<"example.com">>,<<>>,<<"user_x">>,<<"example.com">>,<<>>},{jid,<<"user_x">>,<<"example.com">>,<<"14431102907355708286168">>,<<"user_x">>,<<"example.com">>,<<"14431102907355708286168">>},{iq,<<"purplebb730878">>,result,<<>>,{jid,<<"user_x">>,<<"example.com">>,<<>>,<<"user_x">>,<<"example.com">>,<<>>},{jid,<<"user_x">>,<<"example.com">>,<<"14431102907355708286168">>,<<"user_x">>,<<"example.com">>,<<"14431102907355708286168">>},[{pubsub,undefined,undefined,undefined,undefined,undefined,undefined,undefined,undefined,{ps_items,<<>>,<<"urn:xmpp:avatar:metadata">>,[{ps_item,<<>>,"05af2e5d1baddbcf8b60f84030e418f3c94eb409",[{xmlelement,"metadata",[{"xmlns","urn:xmpp:avatar:metadata"}],[{xmlelement,"info",[{"id","05af2e5d1baddbcf8b60f84030e418f3c94eb409"},{"type","image/png"},{"bytes","5581"},{"width","80"},{"height","80"}],[]}]}],<<>>,<<>>}],undefined,<<>>,undefined},undefined,undefined,undefined,undefined,undefined,undefined,undefined}],#{}}}
** When State == session_established
**      Data  == {state,{socket_state,fast_tls,{tlssock,#Port<0.19262>,#Port<0.19263>},<0.753.0>},ejabberd_socket,#Ref<0.0.1.14149>,false,<<"14186485891198277559">>,undefined,c2s,c2s_shaper,false,true,true,true,[verify_none,compression_none,{protocol_options,<<"no_sslv3|no_sslv2">>},{dhfile,<<"/etc/ssl/private/ejabberd/dh2048.pem">>},{certfile,<<"/etc/ssl/private/wildcard.example.com-with-intermediate-and-key.crt">>},{ciphers,<<"ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES256-SHA384:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES256-SHA384:ECDHE-ECDSA-AES256-SHA:ECDHE-RSA-AES256-SHA:AES128-GCM-SHA256:AES256-GCM-SHA384:AES128-SHA256:AES256-SHA256:AES128-SHA:AES256-SHA:!DSS">>}],true,{jid,<<"user_x">>,<<"example.com">>,<<"14431102907355708286168">>,<<"user_x">>,<<"example.com">>,<<"14431102907355708286168">>},<<"user_x">>,<<"example.com">>,<<"14431102907355708286168">>,{{1483,421863,2},<0.754.0>},{pres_t,13},{pres_f,14},{pres_a,0},undefined,undefined,{userlist,<<"Blocked contacts">>,[],false},c2s_tls,ejabberd_auth_mnesia,{{74,94,156,211},52124},[],active,inactive,undefined,undefined,1000,undefined,300,300,60000,undefined,false,0,0,0,true,<<>>}
** Reason for termination = 
** {badarg,[{fxml,element_to_binary,[{xmlel,<<"iq">>,[{<<"to">>,<<"user_x@example.com/14431102907355708286168">>},{<<"from">>,<<"user_x@example.com">>},{<<"type">>,<<"result">>},{<<"id">>,<<"purplebb730878">>}],[{xmlel,<<"pubsub">>,[{<<"xmlns">>,<<"http://jabber.org/protocol/pubsub">>}],[{xmlel,<<"items">>,[{<<"node">>,<<"urn:xmpp:avatar:metadata">>}],[{xmlel,<<"item">>,[{<<"id">>,"05af2e5d1baddbcf8b60f84030e418f3c94eb409"}],[{xmlelement,"metadata",[{"xmlns","urn:xmpp:avatar:metadata"}],[{xmlelement,"info",[{"id","05af2e5d1baddbcf8b60f84030e418f3c94eb409"},{"type","image/png"},{"bytes","5581"},{"width","80"},{"height","80"}],[]}]}]}]}]}]}],[]},{ejabberd_c2s,send_element,2,[{file,"src/ejabberd_c2s.erl"},{line,1649}]},{ejabberd_c2s,send_stanza,2,[{file,"src/ejabberd_c2s.erl"},{line,1672}]},{ejabberd_c2s,handle_info,3,[{file,"src/ejabberd_c2s.erl"},{line,1427}]},{p1_fsm,handle_msg,10,[{file,"src/p1_fsm.erl"},{line,582}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,240}]}]}
2017-01-03 00:37:45 =CRASH REPORT====
  crasher:
    initial call: gen:init_it/6
    pid: <0.754.0>
    registered_name: []
    exception exit: {{badarg,[{fxml,element_to_binary,[{xmlel,<<"iq">>,[{<<"to">>,<<"user_x@example.com/14431102907355708286168">>},{<<"from">>,<<"user_x@example.com">>},{<<"type">>,<<"result">>},{<<"id">>,<<"purplebb730878">>}],[{xmlel,<<"pubsub">>,[{<<"xmlns">>,<<"http://jabber.org/protocol/pubsub">>}],[{xmlel,<<"items">>,[{<<"node">>,<<"urn:xmpp:avatar:metadata">>}],[{xmlel,<<"item">>,[{<<"id">>,"05af2e5d1baddbcf8b60f84030e418f3c94eb409"}],[{xmlelement,"metadata",[{"xmlns","urn:xmpp:avatar:metadata"}],[{xmlelement,"info",[{"id","05af2e5d1baddbcf8b60f84030e418f3c94eb409"},{"type","image/png"},{"bytes","5581"},{"width","80"},{"height","80"}],[]}]}]}]}]}]}],[]},{ejabberd_c2s,send_element,2,[{file,"src/ejabberd_c2s.erl"},{line,1649}]},{ejabberd_c2s,send_stanza,2,[{file,"src/ejabberd_c2s.erl"},{line,1672}]},{ejabberd_c2s,handle_info,3,[{file,"src/ejabberd_c2s.erl"},{line,1427}]},{p1_fsm,handle_msg,10,[{file,"src/p1_fsm.erl"},{line,582}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,240}]}]},[{p1_fsm,terminate,8,[{file,"src/p1_fsm.erl"},{line,760}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,240}]}]}
    ancestors: [<0.664.0>,ejabberd_listeners,ejabberd_sup,<0.37.0>]
    messages: []
    links: [#Port<0.19263>]
    dictionary: [{'$internal_queue_len',0}]
    trap_exit: false
    status: running
    heap_size: 196650
    stack_size: 27
    reductions: 312920
  neighbours:
````

**Comment by sudoman** [comment]
This may be related to  #1486.

**Comment by zinid** [comment]
Fixed in 17.01

**Comment by sudoman** [comment]
For anyone who stumbles upon this issue, here is the db upgrade method in case you want to go back to using the version of ejabberd in your distro: https://github.com/processone/ejabberd/issues/1409#issuecomment-271549380
